modules/dataplex-datascan/variables.tf

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "data" {
+  description = "The data source for DataScan. The source can be either a Dataplex `entity` or a BigQuery `resource`."
+  type = object({
+    entity   = optional(string)
+    resource = optional(string)
+  })
+  validation {
+    condition     = length([for k, v in var.data : v if contains(["resource", "entity"], k) && v != null]) == 1
+    error_message = "Datascan data must specify one of 'entity', 'resource'."
+  }
+}
+
+variable "data_profile_spec" {
+  description = "DataProfileScan related setting. Variables description are provided in https://cloud.google.com/dataplex/docs/reference/rest/v1/DataProfileSpec."
```

Comment:
Nit:" `variables description is` or `variable descriptions are` :) Same in other variables.

Reply:
Good catch!